src/hooks/useRoles.ts

```diff
@@ -1,48 +1,60 @@
-import { useFirestoreConnect } from "./useFirestoreConnect";
+import { isEqual } from "lodash";
+
+import { isLoaded, useFirestoreConnect } from "./useFirestoreConnect";
 import { useUser } from "./useUser";
 import { useSelector } from "./useSelector";
 
 export const useRoles = () => {
   const { user } = useUser();
-  useFirestoreConnect({
-    collection: "roles",
-    where: [["users", "array-contains", user?.uid || ""]],
-    storeAs: "userRoles",
-  });
+
+  useFirestoreConnect(
+    user
+      ? {
+          collection: "roles",
+          where: [["users", "array-contains", user.uid]],
+          storeAs: "userRoles",
+        }
+      : undefined
+  );
+
   useFirestoreConnect({
     collection: "roles",
     where: [["allowAll", "==", true]],
     storeAs: "allowAllRoles",
   });
-  const { userRoles, allowAllRoles } = useSelector((state) => ({
-    userRoles: state.firestore.data.userRoles,
-    allowAllRoles: state.firestore.data.allowAllRoles,
-  }));
-
-  // Note: null here means data is loaded, but there was none.
-  // A value of undefined indicates data is not loaded yet.
-  // undefined should be returned so callers can show loading indications
+
+  const {
+    isUserRolesLoaded,
+    isAllowAllRolesLoaded,
+    isRolesLoaded,
+    userRoles,
+    allowAllRoles,
+    roles,
+  } = useSelector((state) => {
+    const { userRoles, allowAllRoles } = state.firestore.data;
+
+    const rolesForUser = userRoles ? Object.keys(userRoles) : [];
+    const rolesForAll = allowAllRoles ? Object.keys(allowAllRoles) : [];
+    const combinedRoles = [...rolesForUser, ...rolesForAll];
+
+    return {
+      isUserRolesLoaded: isLoaded(userRoles),
+      userRoles: rolesForUser,
+
+      isAllowAllRolesLoaded: isLoaded(allowAllRoles),
+      allowAllRoles: rolesForAll,
+
+      isRolesLoaded: isLoaded(userRoles) && isLoaded(allowAllRoles),
+      roles: combinedRoles,
+    };
```

Comment:
This can actually break some of the logic. As far as I could understand, it would return userRoles/allowAllRoles as undefined if it was undefined and as an empty array if it was null. (NOTE: null means no data, undefined means data not loaded in Firestore language)

Reply:
The new change will return empty array for both `undefined` and `null`

Reply:
Yeah, but I went and looked around at all the usages and IMO nowhere even used it. Would be good for you to check and confirm whether you agree with that.

Reply:
Confirming that it won't break anything. A bit weird piece in 
![image](https://user-images.githubusercontent.com/63194656/125476207-4b3ee753-83d6-4431-b707-010af65f78e7.png)

@0xdevalias 
NOTE: It seems like all the places where it's used just needs `isAdmin` boolean

Reply:
> NOTE: It seems like all the places where it's used just needs isAdmin boolean

Yeah, that's pretty much what I observed as well.

Reply:
That `useRoles` + `useIsAdminUser` is probably people not understanding the hooks, as I believe they both read from the exact same place (though via separate firestore queries)

Reply:
Okay, then I suppose it's OOS